sbin/common/config_init.sh

```diff
@@ -270,8 +270,14 @@ function parseConfigurationArguments() {
         "--destination" | "-d" )
         BUILD_CONFIG[TARGET_DIR]="$1"; shift;;
 
-        "--docker" | "-D" )
-        BUILD_CONFIG[USE_DOCKER]="true";;
+        "-D" )
```

Comment:
What is the reason for using this `USE_DOCKER` variable instead of the exsting `DOCKER` one. It means we now have lines elsewhere like:

`${BUILD_CONFIG[DOCKER]} ${BUILD_CONFIG[USE_DOCKER]}` which aren't as clear as they could be when you look at those lines in isolation.

Reply:
Yes they are absolutely misleading, but they were already before. Before this goes in, I would like to rename ethem both, so they  fit at least a bit to what they are doing.  I have left them in for readability (although after all, it do not help.)

Originally the         `BUILD_CONFIG[USE_DOCKER] was true/false` and` ${BUILD_CONFIG[DOCKER]} was "docker" or "sudo docker"`.   Now the `BUILD_CONFIG[USE_DOCKER] is docker/podman` and `${BUILD_CONFIG[DOCKER]} is sudo or nothing`

 The rname should be  `BUILD_CONFIG[CONTAINER_COMMAND]` and `${BUILD_CONFIG[SUDO_CONTAINER]} `

Reply:
Actually I was long hesitating wheter to keep using `USE_DOCKER `true/false flag,  and keep command in `DOCKER` with  `docker/podman/sudo docker/sudo podman` and decided with slight 55:45 overwhelm to use  USE_DOCKER as command and DOCKER as sudo preffix, with intention to rename at the end.

Reply:
@andrew-m-leonard Am I correct in saying that thse parameters are not defined at any point in the jenkins jobs so changing them here should be "safe"?

Reply:
Actually - main reason to swithc, was to add more flexibility for possible "sudo" fixes. Eg to allow choice sudo/run0/doas ....

Reply:
> @andrew-m-leonard Am I correct in saying that thse parameters are not defined at any point in the jenkins jobs so changing them here should be "safe"?


Great remark. No idea. Then the rename will go as spearte PR onc.. if... those changese ever goes in. TYVM!

Reply:
ping @andrew-m-leonard ;-)

Reply:
> Actually I was long hesitating wheter to keep using USE_DOCKER true/false flag, and keep command in DOCKER with docker/podman/sudo docker/sudo podman and decided with slight 55:45 overwhelm to use USE_DOCKER as command and DOCKER as sudo preffix, with intention to rename at the end.

Yeah makes sense - I think if we do the renames so that it's a little more comprehensible to new people looking at the scripts.

```
${BUILD_CONFIG[DOCKER]} "${BUILD_CONFIG[USE_DOCKER]}"
```
Looks a bit odd in isolation so something like
```
${BUILD_CONFIG[SUDO_CONTAINER]} BUILD_CONFIG[CONTAINER_COMMAND]
```
 would be a lot clearer, so if we can get some of these renames in (and hope this PR rebases on master nicely!) then I'm personally good with putting this in if Andrew agrees.

Reply:
Right. The renaming to somethnig like that will be necessary. Currently it is terribly unreadable.
yes, the renaming is missing due to rebases :) .. no it do not rebase smoothly, but I will elaborate !
`CONTAINER_COMMAND` sounds like clear choice.  `CONTAINER_AS_ROOT` may be better if we want to prevent another renaming in future (and `PRIVLIDGED_CONTAINER` is jsut wrong).   But currently it is indeed jsut sudo. So `SUDO_CONTAINER`  would do.